luavm/src/lib/stdlib.rs

```diff
@@ -1,38 +1,65 @@
+use super::errors::LuaError;
 use super::Vm;
 use std::fmt::Write as FmtWrite;
 
-pub const STDLIB_FUNCS: &'static [StdFunction] = &[StdFunction {
-    name: "print",
-    handler: lua_print,
-    param_count: 0,
-}];
+pub const STDLIB_FUNCS: &'static [StdFunction] = &[
+    StdFunction {
+        name: "assert",
+        handler: lua_assert,
+    },
+    StdFunction {
+        name: "print",
+        handler: lua_print,
+    },
+];
 
 pub struct StdFunction {
     name: &'static str,
-    handler: fn(&mut Vm),
-    param_count: usize,
+    handler: fn(&mut Vm) -> Result<(), LuaError>,
 }
 
 impl StdFunction {
     pub fn name(&self) -> &str {
         &self.name
     }
 
-    pub fn handler(&self) -> fn(&mut Vm) {
+    pub fn handler(&self) -> fn(&mut Vm) -> Result<(), LuaError> {
         self.handler.clone()
     }
-
-    pub fn param_count(&self) -> usize {
-        self.param_count
-    }
 }
 
-pub fn lua_print(vm: &mut Vm) {
+pub fn lua_print(vm: &mut Vm) -> Result<(), LuaError> {
     let args_start = vm.closure.args_start();
     let args_count = vm.closure.args_count();
     let mut s = String::new();
     for i in args_start..(args_start + args_count - 1) {
         write!(s, "{}\t", &vm.stack[i]).unwrap();
     }
     println!("{}{}", s, &vm.stack[args_start + args_count - 1]);
+    Ok(())
+}
+
+pub fn lua_assert(vm: &mut Vm) -> Result<(), LuaError> {
+    let args_start = vm.closure.args_start();
+    let args_count = vm.closure.args_count();
+    if args_count == 0 {
+        return Err(LuaError::Error(
+            "assert expects at least one argument!".to_string(),
+        ));
+    }
+    if vm.stack[args_start].to_bool() {
+        for i in args_start..(args_start + args_count) {
```

Comment:
https://www.lua.org/manual/5.3/manual.html#pdf-assert

The assert function returns all of its arguments, thus I push all the arguments again to the stack.

Reply:
All its arguments == 1 or 2 arguments and no more?

Reply:
In Lua you can pass as many arguments as you want to a function. In this case, if you call `assert(1 == 1, "foo", 1, 2, 3)` this will return `true, "foo", 1, 2, 3`. I am trying to mimic this functionality.

Reply:
Got it.